machine/v1beta1/types_gcpprovider.go

```diff
@@ -149,6 +149,49 @@ type GCPMachineProviderSpec struct {
 	// +kubebuilder:validation:Enum=Enabled;Disabled
 	// +optional
 	ConfidentialCompute ConfidentialComputePolicy `json:"confidentialCompute,omitempty"`
+
+	// resourceManagerTags is an optional list of tags to apply to the GCP resources created for
+	// the cluster. See https://cloud.google.com/resource-manager/docs/tags/tags-overview for
+	// information on tagging GCP resources. GCP supports a maximum of 50 tags per resource.
+	// +kubebuilder:validation:MaxItems=50
+	// +listType=map
+	// +listMapKey=key
+	// +optional
+	ResourceManagerTags []ResourceManagerTag `json:"resourceManagerTags,omitempty"`
+}
+
+// ResourceManagerTag is a tag to apply to GCP resources created for the cluster.
+type ResourceManagerTag struct {
+	// parentID is the ID of the hierarchical resource where the tags are defined
+	// e.g. at the Organization or the Project level. To find the Organization or Project ID ref
+	// https://cloud.google.com/resource-manager/docs/creating-managing-organization#retrieving_your_organization_id
+	// https://cloud.google.com/resource-manager/docs/creating-managing-projects#identifying_projects
+	// An OrganizationID must consist of decimal numbers, and cannot have leading zeroes.
+	// A ProjectID must be 6 to 30 characters in length, can only contain lowercase letters, numbers,
```

Comment:
This comment doesn't align with the min/max length validation, which is correct?

Reply:
ParentID can be either organization or project ID. Organization ID must be 1-32 characters in length and Project ID must be 6-30 characters in length. Considering both I kept min and max to 1 and 32. Regex pattern has the additional check for length and the character set allowed for each ID. Please let me know WDYT?

Reply:
This would be clearer as a discriminated union. How does the code know whether it's an organisation or a project ID?

If we are keeping this way, we need to include the length limit for the organisation ID within the description, as you have done for project ID already

Reply:
Code doesn't distinguish the ID, it fetches tag details in the provided namespace => parentID.
I will update the description with the length limit.

Reply:
suggestion incorporated.